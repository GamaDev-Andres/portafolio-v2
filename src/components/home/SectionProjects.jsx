import { proyectos } from '../../helpers/data'
import TitleSection from '../TitleSection'

import CardProject from './CardProject'
const SectionProjects = () => {
  return (
    <section id="proyectos">
      <TitleSection textInSpan={'proyectos'} unOrder={true}>
        mis
      </TitleSection>
      <div className="grid grid-cols-1 md:grid-cols-layoutCards gap-8">
        {proyectos.map((el) => (
          <CardProject project={el} key={el.titulo} />
        ))}
      </div>
    </section>
  )
}

export default SectionProjects
