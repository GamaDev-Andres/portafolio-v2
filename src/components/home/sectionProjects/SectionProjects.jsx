import TitleSection from '../../TitleSection'

import ListOfCardsProjects from './ListOfCardsProjects'
const SectionProjects = () => {
  return (
    <section id="proyectos">
      <TitleSection textInSpan={'proyectos'} unOrder={true}>
        mis
      </TitleSection>
      <ListOfCardsProjects />
    </section>
  )
}

export default SectionProjects
