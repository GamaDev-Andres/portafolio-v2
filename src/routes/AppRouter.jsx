import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
