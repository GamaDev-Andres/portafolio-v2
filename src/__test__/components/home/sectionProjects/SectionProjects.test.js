/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import SectionProjects from '../../../../components/home/sectionProjects/SectionProjects'
import AppRouter from '../../../../routes/AppRouter'

describe('probando <SectionProjects/>', () => {
  const wrapper = Enzyme.mount(
    <MemoryRouter>
      <SectionProjects />
    </MemoryRouter>
  )
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('TitleSection').length).toBe(1)
    expect(wrapper.find('ListOfCardsProjects').exists()).toBe(true)
  })
  test('No debe montarse en ruta "project"', () => {
    const wrapper = Enzyme.mount(
      <HelmetProvider>
        <MemoryRouter initialEntries={['/project/BARBERAPP']}>
          <AppRouter />
        </MemoryRouter>
      </HelmetProvider>
    )
    expect(wrapper.find('SectionProjects').exists()).toBe(false)
  })
})
