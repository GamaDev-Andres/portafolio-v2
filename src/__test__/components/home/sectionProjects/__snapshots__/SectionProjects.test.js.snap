// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <SectionProjects/> debe montarse correctamente 1`] = `
<section
  className="sm:scroll-m-24 md:scroll-m-14"
  id="proyectos"
>
  <h2
    className="my-4 uppercase font-bold text-3xl md:text-center"
  >
    mis
     
    <span
      className="text-blue-400"
    >
      proyectos
    </span>
  </h2>
  <div
    className="grid grid-cols-1 md:grid-cols-layoutCards gap-8"
  >
    <div
      className=" shadow-blue-200 shadow-sm rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos"
    >
      <a
        className="w-full block aspect-video"
        href="https://insta-gama.vercel.app/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="project"
          className="w-full aspect-video object-cover"
          src="https://res.cloudinary.com/dapa84kxy/image/upload/v1652509941/portafolio/instaGama/Image_1_n55ffj.png"
        />
      </a>
      <div
        className="px-4 pb-4"
      >
        <div
          className="relative py-2 flex justify-center items-center"
        >
          <a
            className="font-bold text-blue-400 text-center mx-auto text-xl"
            href="https://insta-gama.vercel.app/"
            rel="noreferrer"
            target="_blank"
          >
            INSTAGAMA
             
            <i
              className="text-sm fa-solid fa-link dark:text-letraDark text-fondos"
            />
          </a>
          <a
            className="absolute top-2 right-0 text-xl block text-letra"
            href="https://github.com/GamaDev-Andres/instaGama"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="dark:text-letraDark fa-brands fa-github"
            />
          </a>
        </div>
        <div>
          <h3
            className=" dark:text-letraDark text-letra font-medium"
          >
            Descripción
          </h3>
          <p>
            Proyecto con objetivo académico y practico , en donde creo un clon funcional de Instagram con sus funcionalidades básicas. implementando tecnologías como vite, MERN, sockets entre otras.
          </p>
        </div>
        <div>
          <h3
            className="text-letra font-medium dark:text-letraDark"
          >
            Tecnologias implementadas
          </h3>
          <ul
            className="flex gap-2 flex-wrap py-2"
          >
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="React"
            >
              <span>
                React
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Node"
            >
              <span>
                Node
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Express"
            >
              <span>
                Express
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="MongoDb"
            >
              <span>
                MongoDb
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Tailwindcss"
            >
              <span>
                Tailwindcss
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Sockets.io"
            >
              <span>
                Sockets.io
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Cloudinary"
            >
              <span>
                Cloudinary
              </span>
            </li>
          </ul>
        </div>
        <a
          className=" hover:shadow-sm hover:shadow-blue-200 duration-300 text-sm text-center mx-auto block mt-4 border border-solid border-blue-400 px-6 py-2 rounded-full"
          href="/project/INSTAGAMA"
          onClick={[Function]}
          rel="noreferrer"
        >
          Ver más detalles
        </a>
      </div>
    </div>
    <div
      className=" shadow-blue-200 shadow-sm rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos"
    >
      <a
        className="w-full block aspect-video"
        href="https://barberappag.netlify.app/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="project"
          className="w-full aspect-video object-cover"
          src="https://res.cloudinary.com/dapa84kxy/image/upload/v1644874961/portafolio/barberapp/barberapp_sdpgwl.png"
        />
      </a>
      <div
        className="px-4 pb-4"
      >
        <div
          className="relative py-2 flex justify-center items-center"
        >
          <a
            className="font-bold text-blue-400 text-center mx-auto text-xl"
            href="https://barberappag.netlify.app/"
            rel="noreferrer"
            target="_blank"
          >
            BARBERAPP
             
            <i
              className="text-sm fa-solid fa-link dark:text-letraDark text-fondos"
            />
          </a>
          <a
            className="absolute top-2 right-0 text-xl block text-letra"
            href="https://github.com/GamaDev-Andres/BarberApp-fe"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="dark:text-letraDark fa-brands fa-github"
            />
          </a>
        </div>
        <div>
          <h3
            className=" dark:text-letraDark text-letra font-medium"
          >
            Descripción
          </h3>
          <p>
            Aplicación realizada con el stack MERN y styled components,
     enfocada en la administración de una barbería en donde el cliente tiene 
     permitido agendar una cita con un barbero basándose en su perfil ,
      cortes o calificación de otros clientes.
          </p>
        </div>
        <div>
          <h3
            className="text-letra font-medium dark:text-letraDark"
          >
            Tecnologias implementadas
          </h3>
          <ul
            className="flex gap-2 flex-wrap py-2"
          >
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="React"
            >
              <span>
                React
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Express"
            >
              <span>
                Express
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="StyledComponents"
            >
              <span>
                StyledComponents
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="MongoDb"
            >
              <span>
                MongoDb
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Reducers"
            >
              <span>
                Reducers
              </span>
            </li>
          </ul>
        </div>
        <a
          className=" hover:shadow-sm hover:shadow-blue-200 duration-300 text-sm text-center mx-auto block mt-4 border border-solid border-blue-400 px-6 py-2 rounded-full"
          href="/project/BARBERAPP"
          onClick={[Function]}
          rel="noreferrer"
        >
          Ver más detalles
        </a>
      </div>
    </div>
    <div
      className=" shadow-blue-200 shadow-sm rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos"
    >
      <a
        className="w-full block aspect-video"
        href="https://agnetbook.netlify.app/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="project"
          className="w-full aspect-video object-cover"
          src="https://res.cloudinary.com/dapa84kxy/image/upload/v1644874998/portafolio/netbook/netbook_gus84s.png"
        />
      </a>
      <div
        className="px-4 pb-4"
      >
        <div
          className="relative py-2 flex justify-center items-center"
        >
          <a
            className="font-bold text-blue-400 text-center mx-auto text-xl"
            href="https://agnetbook.netlify.app/"
            rel="noreferrer"
            target="_blank"
          >
            NETBOOK
             
            <i
              className="text-sm fa-solid fa-link dark:text-letraDark text-fondos"
            />
          </a>
          <a
            className="absolute top-2 right-0 text-xl block text-letra"
            href="https://github.com/GamaDev-Andres/network-social-react"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="dark:text-letraDark fa-brands fa-github"
            />
          </a>
        </div>
        <div>
          <h3
            className=" dark:text-letraDark text-letra font-medium"
          >
            Descripción
          </h3>
          <p>
            Mini clon de una red social realizado con React, Redux y firebase, me baso en ciertas funcionalidades de Facebook, tales como, publicaciones, likes, comentarios y perfiles individuales.
          </p>
        </div>
        <div>
          <h3
            className="text-letra font-medium dark:text-letraDark"
          >
            Tecnologias implementadas
          </h3>
          <ul
            className="flex gap-2 flex-wrap py-2"
          >
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="React"
            >
              <span>
                React
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Firebase"
            >
              <span>
                Firebase
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Redux"
            >
              <span>
                Redux
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Sass"
            >
              <span>
                Sass
              </span>
            </li>
          </ul>
        </div>
        <a
          className=" hover:shadow-sm hover:shadow-blue-200 duration-300 text-sm text-center mx-auto block mt-4 border border-solid border-blue-400 px-6 py-2 rounded-full"
          href="/project/NETBOOK"
          onClick={[Function]}
          rel="noreferrer"
        >
          Ver más detalles
        </a>
      </div>
    </div>
    <div
      className=" shadow-blue-200 shadow-sm rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos"
    >
      <a
        className="w-full block aspect-video"
        href="https://practica-galeria.web.app/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="project"
          className="w-full aspect-video object-cover"
          src="https://res.cloudinary.com/dapa84kxy/image/upload/v1644875042/portafolio/galeriareact/reactFirebase_xjiauw.png"
        />
      </a>
      <div
        className="px-4 pb-4"
      >
        <div
          className="relative py-2 flex justify-center items-center"
        >
          <a
            className="font-bold text-blue-400 text-center mx-auto text-xl"
            href="https://practica-galeria.web.app/"
            rel="noreferrer"
            target="_blank"
          >
            GALERIA
             
            <i
              className="text-sm fa-solid fa-link dark:text-letraDark text-fondos"
            />
          </a>
          <a
            className="absolute top-2 right-0 text-xl block text-letra"
            href="https://github.com/GamaDev-Andres/react-galeria"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="dark:text-letraDark fa-brands fa-github"
            />
          </a>
        </div>
        <div>
          <h3
            className=" dark:text-letraDark text-letra font-medium"
          >
            Descripción
          </h3>
          <p>
            Simulación de una galería web realizada con React y Firebase , la aplicación se basa en la posibilidad de crear álbumes y tomarse fotos directamente en el álbum deseado y también tiene la posibilidad de buscar imágenes en una API y guardarlas en un álbum.
          </p>
        </div>
        <div>
          <h3
            className="text-letra font-medium dark:text-letraDark"
          >
            Tecnologias implementadas
          </h3>
          <ul
            className="flex gap-2 flex-wrap py-2"
          >
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="React"
            >
              <span>
                React
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Firebase"
            >
              <span>
                Firebase
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="StyledComponents"
            >
              <span>
                StyledComponents
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className=" shadow-blue-200 shadow-sm rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos"
    >
      <a
        className="w-full block aspect-video"
        href="https://afmispelisgi.netlify.app/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="project"
          className="w-full aspect-video object-cover"
          src="https://res.cloudinary.com/dapa84kxy/image/upload/v1644875064/portafolio/mispelis/imagepelisOMDB_cchblf.png"
        />
      </a>
      <div
        className="px-4 pb-4"
      >
        <div
          className="relative py-2 flex justify-center items-center"
        >
          <a
            className="font-bold text-blue-400 text-center mx-auto text-xl"
            href="https://afmispelisgi.netlify.app/"
            rel="noreferrer"
            target="_blank"
          >
            MIS PELIS
             
            <i
              className="text-sm fa-solid fa-link dark:text-letraDark text-fondos"
            />
          </a>
          <a
            className="absolute top-2 right-0 text-xl block text-letra"
            href="https://github.com/GamaDev-Andres/PeliculasOMDB"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="dark:text-letraDark fa-brands fa-github"
            />
          </a>
        </div>
        <div>
          <h3
            className=" dark:text-letraDark text-letra font-medium"
          >
            Descripción
          </h3>
          <p>
            Proyecto de películas hecho con Javascript, Sass y webpack .En este proyecto me baso en un diseño de una pagina funcional, consulto la API IMDB que me provee información tanto de series como de películas con sus respectivos tráileres y sinopsis. Implemento buscadores y lista de favoritos.
          </p>
        </div>
        <div>
          <h3
            className="text-letra font-medium dark:text-letraDark"
          >
            Tecnologias implementadas
          </h3>
          <ul
            className="flex gap-2 flex-wrap py-2"
          >
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Html"
            >
              <span>
                Html
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Sass"
            >
              <span>
                Sass
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Webpack"
            >
              <span>
                Webpack
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Javascript"
            >
              <span>
                Javascript
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className=" shadow-blue-200 shadow-sm rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos"
    >
      <a
        className="w-full block aspect-video"
        href="https://findyourgames.netlify.app/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="project"
          className="w-full aspect-video object-cover"
          src="https://res.cloudinary.com/dapa84kxy/image/upload/v1644875081/portafolio/findyourgames/proyecto1_vsncjy.png"
        />
      </a>
      <div
        className="px-4 pb-4"
      >
        <div
          className="relative py-2 flex justify-center items-center"
        >
          <a
            className="font-bold text-blue-400 text-center mx-auto text-xl"
            href="https://findyourgames.netlify.app/"
            rel="noreferrer"
            target="_blank"
          >
            FINDYOURGAMES
             
            <i
              className="text-sm fa-solid fa-link dark:text-letraDark text-fondos"
            />
          </a>
          <a
            className="absolute top-2 right-0 text-xl block text-letra"
            href="https://github.com/GamaDev-Andres/FindYourGames"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="dark:text-letraDark fa-brands fa-github"
            />
          </a>
        </div>
        <div>
          <h3
            className=" dark:text-letraDark text-letra font-medium"
          >
            Descripción
          </h3>
          <p>
            Proyecto realizado con Html, css, javascript y gulp. En este proyecto consulto una API de videojuegos y organizo su información en una interfaz que me permite hacer búsqueda y filtrado de los videojuegos disponibles.
          </p>
        </div>
        <div>
          <h3
            className="text-letra font-medium dark:text-letraDark"
          >
            Tecnologias implementadas
          </h3>
          <ul
            className="flex gap-2 flex-wrap py-2"
          >
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Html"
            >
              <span>
                Html
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Sass"
            >
              <span>
                Sass
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Gulp"
            >
              <span>
                Gulp
              </span>
            </li>
            <li
              className="px-2 py-1 border border-bordes rounded-full"
              key="Javascript"
            >
              <span>
                Javascript
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</section>
`;
